source/gfoidl.Base64/Internal/Base64Encoder/Base64Encoder.Decode.netcoreapp.cs

```diff
@@ -39,21 +40,26 @@ public partial class Base64Encoder
 
             ref byte destBytes = ref MemoryMarshal.GetReference(data);
 
+            if (maxSrcLength < 24)
+                goto Scalar;
+
             if (Avx2.IsSupported && maxSrcLength >= 45)
             {
                 Avx2Decode(ref src, ref destBytes, maxSrcLength, destLength, ref sourceIndex, ref destIndex);
 
                 if (sourceIndex == srcLength)
                     goto DoneExit;
             }
-            else if (Ssse3.IsSupported && maxSrcLength >= 24)
+
+            if (Ssse3.IsSupported && (maxSrcLength >= (int)sourceIndex + 24))
```

Comment:
Verify perf of this change. Back then perf was worse, so this was intentional.

Reply:
Will be investigated in https://github.com/gfoidl/Base64/issues/96

Reply:
There is a bug anyway, as AVX2 and SSSE3 get executed. 
Showed up in a benchmark where with increasing size the gap got bigger and bigger.

Reply:
Bug will be fixed in https://github.com/gfoidl/Base64/pull/98